manager/controllers/app/blueprint_controller.go

```diff
@@ -163,26 +166,53 @@ func (r *BlueprintReconciler) reconcile(ctx context.Context, log logr.Logger, bl
 		// Add metadata arguments (namespace, name, labels, etc.)
 		args["metadata"] = map[string]interface{}{
 			"name":      utils.Hash(step.Name, 20),
-			"namespace": blueprint.Namespace, // TODO(roee.shlomo): should eventually be a dedicated namespace per M4DApplication
+			"namespace": blueprint.Namespace,
 			"labels":    blueprint.Labels,
 		}
+		// we add the "r" character at the beginning of the release name, since it must begin with an alphabetic character
+		releaseName := "r" + args["metadata"].(map[string]interface{})["name"].(string)
+		log.V(0).Info("Release name: " + releaseName)
+		// check if the blueprint is about to be deleted
+		if blueprint.DeletionTimestamp != nil {
+			_, _ = r.deleteChartResource(ctx, blueprint.Namespace, releaseName)
```

Comment:
Shouldn't this be done in a finalizer?

Reply:
It should. Currently, blueprints do not have finalizers, and the chart deletion existed in master as well. I remember an old Trello task regarding this, will open a new issue and a create a new PR.